### PR TITLE
 Docs/rearranging top level pages

### PR DESCRIPTION
**Pull request by jacalata**
## Summary of changes
Deleted setuptools.rst and redirected incoming links to index page
- dropped Features list and Transition Note
- moved setup.cfg-only section to setup.cfg page
- moved Forum and Bug Tracker to index page
- moved manifest note to data files page
- moved Configuration API to Deprecated Functionalities?

Closes https://github.com/pypa/setuptools/discussions/3400

### Pull Request Checklist
- [n/a ] Changes have tests
- [x] News fragment added in [`changelog.d/`].
  _(See [documentation][PR docs] for details)_

[`changelog.d/`]: https://github.com/pypa/setuptools/tree/master/changelog.d
[PR docs]:
https://setuptools.pypa.io/en/latest/development/developer-guide.html#making-a-pull-request

### CODE REVIEW

**Comment by jacalata** [comment]
test failure is unrelated, waiting for fix in https://github.com/pypa/setuptools/pull/3410

**Comment by abravalheri** [comment]
Yes, thank you very much @jacalata .

I have some suggestions to make but I still haven't had the time to write them down. I will probably send you a PR soon.

There is one comment about the redirect setup. I remember reading somewhere that the plugin setuptools is using for the redirects requires the dedtination address to have the `.html` extension. Could you please double check that?

File: docs/deprecated/functionalities.rst [hunk 1]
```diff
@@ -1,5 +1,5 @@
 "Eggsecutable" Scripts
-----------------------
+=================
```
Comment (jaraco) [inline docs/deprecated/functionalities.rst:2]: I recall that rst is strict about heading length matches, so will likely warn here (and fail in nitpicky mode). Let's keep the length consistent.

File: docs/userguide/declarative_config.rst [hunk 2]
```diff
@@ -4,19 +4,36 @@
...
+if you **can rely on** your project always being built by a :pep:`517`/:pep:`518`
+compatible frontend. This approach not only allows automation scenarios but
+also reduces boilerplate code in some cases.
+
+If compatibility with legacy builds (i.e. those not using the :pep:`517`
+build API) is desired, a ``setup.py`` file containing a ``setup()`` function
+call is still required even if your configuration resides in ``setup.cfg``/``pyproject.toml``.
+
+
+.. note:: TODO: inserted from "Building and Distributing Packages": is this still true?
```
[the remark this answers is not in the document]
Comment (jacalata) [reply docs/userguide/declarative_config.rst:28]: I was hoping to figure this one out before submitting, it seems to mildly conflict with some other text.

File: docs/deprecated/functionalities.rst (hunk 1 above)
Comment (funkyfuture) [inline docs/deprecated/functionalities.rst:2]:
```suggestion
======================
```

File: docs/userguide/declarative_config.rst [hunk 3]
```diff
@@ -4,19 +4,36 @@
...
+
+	versionadded:: 40.9.0
```
[the remark this answers is not in the document]
Comment (webknjaz) [reply docs/userguide/declarative_config.rst:30]:
```suggestion
	.. versionadded:: 40.9.0
```

**Comment by webknjaz** [comment]
@jacalata looks like this PR needs rebasing

**Comment by jaraco** [comment]
This PR seems stale. I'm closing it, but feel free to engage on it again, including requesting to re-open or by re-submitting (presumably after addressing some or all of the concerns above).

**Pull request closed without merging**